src/GitHub.Exports.Reactive/Collections/TrackingCollection.cs

```diff
@@ -47,53 +48,120 @@ public static class TrackingCollection
             IList<T> stickieItemsOnTop = null)
             where T : class, ICopyable<T>
         {
-            var col = new ObservableCollection<T>(stickieItemsOnTop ?? Enumerable.Empty<T>());
-            tcol.CollectionChanged += (s, e) =>
+            if (stickieItemsOnTop == null)
             {
-                var offset = 0;
-                if (stickieItemsOnTop != null)
+                stickieItemsOnTop = new T[0];
+            }
+
+            var col = new ObservableCollection<T>(stickieItemsOnTop.Concat(tcol));
+            tcol.CollectionChanged += (_, e) => UpdateStickieItems(col, e, stickieItemsOnTop);
+            return col;
+        }
+
+        /// <summary>
+        /// Creates an observable collection that tracks an <see cref="ITrackingCollection{T}"/>
+        /// and adds a sticky item to the top of the collection when a related selection is null.
+        /// </summary>
+        /// <typeparam name="T">The type of items in the collection.</typeparam>
+        /// <param name="tcol">The source tracking collection</param>
+        /// <param name="stickieItemOnTop">The sticky item to add to the top of the collection.</param>
+        /// <param name="selection">
+        /// The current selection. If null or equal to the sticky item then the sticky item will be
+        /// added to the collection.
+        /// </param>
+        /// <returns>An <see cref="ObservableCollection{T}"/>.</returns>
+        public static ObservableCollection<T> CreateListenerCollection<T>(this ITrackingCollection<T> tcol,
+            T stickieItemOnTop,
+            IObservable<T> selection)
+            where T : class, ICopyable<T>
+        {
+            Debug.Assert(stickieItemOnTop != null, "stickieItemOnTop may not be null in CreateListenerCollection");
+            Debug.Assert(selection != null, "selection may not be null in CreateListenerCollection");
+
+            var stickieItems = new[] { stickieItemOnTop };
+            var result = new ObservableCollection<T>(tcol);
+            var addedStickieItem = false;
+            var hasSelection = false;
+
+            tcol.CollectionChanged += (_, e) =>
+            {
+                UpdateStickieItems(result, e, hasSelection ? stickieItems : null);
+            };
+
+            selection.Subscribe(x =>
+            {
+                if (x == null || object.Equals(x, stickieItemOnTop))
                 {
-                    foreach (var item in stickieItemsOnTop)
+                    if (addedStickieItem)
                     {
```

Comment:
How about we just do a `result.FirstOrDefault() == stickieItemOnTop` instead of tracking with a flag? We know it's always going to be either the first or not there at all, and that way we don't have to keep state (and the first element look up is fast and doesn't consume the whole collection). That would work equally well to replace the  `hasSelection`  flag, too.